### PR TITLE
WIP: bibcheck: new plugin - journal_names_var

### PR DESCRIPTION
**Pull request by ksachs**
* update of plugin journal_names
* proper treatment of dots
* all variants normalized
* short names (711) in variants (730)
* variants are not repeated
* codens and variants not in other record
* option to sort variants

Signed-off-by: Kirsten Sachs <sachs@l00lnxkaos.desy.de>

### CODE REVIEW

**Comment by ksachs** [comment]
An alternative to  https://github.com/inspirehep/inspire/pull/427 to update the existing plugin journal_names.
Fully tested.

Out of 3633 journal records
844   records: short names will be added to variants
3615  records: full names are not in variants, so we leave it like this.
25    records: variants exists in other record

Since more variants will be created this has to run twice to check the new variants for uniqueness.

**Comment by tsgit** [comment]
when run this will create more than 230 tickets in RT for the "invalid" warnings. is that the intention?

**Comment by ksachs** [comment]
it's 1 ticket per rule. So it will create one (long) ticket.

**Comment by tsgit** [comment]
my test run created 400 emails to me

File: bibcheck/plugins/journal_names_var.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+exactly one short name (711__a)
+normalized short name is in name variants (730__a)
+all codens (030__a) and name variants are unique
+Option sort_variants will sort 730 fields by length.
+"""
+def normalize_name(name):
+    """No dots, all-caps"""
+    return name.replace('. ', ' ').replace('.', ' ').strip().upper()
+
+def sorted_keys(dd):
```
Comment (tsgit) [inline bibcheck/plugins/journal_names_var.py]: [k for k, _ in sorted(dd.items(), key=itemgetter(1))]

or use lambda in the key function if you don't want to `from operator import itemgetter`

this is more stable, works in python3, doesn't require the values to be of equal lenght

**Comment by ksachs** [comment]
> my test run created 400 emails to me

sorry!
777539 	bibcheck:weekly runs with '--ticket-creation-policy=per-rule'
I never got more than one ticket per week.

File: bibcheck/plugins/journal_names_var.py (hunk 1 above)
Comment (ksachs) [reply bibcheck/plugins/journal_names_var.py]: Thanks!
This is a line I don't understand - way beyond my Fortran-based programming skills.
But it seams to do what want.

**Pull request merged**